### ISSUE TITLE
No data where added to the excel sheet when I used json_to_sheet function.

### ISSUE THREAD

**Issue by basharAl-zghayar** [SheetJS/sheetjs#1697]
I've got an array of objects "csvData" where each object is loaded from the API with map function, and then I call this function to add the values in the array to the excel sheet like this:
` downloadExcelSheet = () => {
    let wb = XLSX.utils.book_new();

    if (
      this.props.timesheetUsersStore!.csvData !== undefined 
    ) {
      let ws = this.props.timesheetUsersStore!.csvData;
      console.log('data', ws);
      var ws1 = XLSX.utils.json_to_sheet(ws, {
        header: [
          'ID',
          'Full_Name',
          'Status',
          'Hourly_Rate',
          'Total_Hours',
          'Bouns',
          'Total_Salary',
        ],
        skipHeader: false,
      });
      XLSX.utils.book_append_sheet(wb, ws1, 'Time Sheets');
      XLSX.writeFile(wb, 'Name.xlsx');
    }`
but the data where not added to the excel when I call the function later.
when I preview the "ws" values in the console I got proxy values which is correct, and when I use `console.log(JSON.stringify(ws))` 
I see the in the console the objects are correctly appears.
 But when the csvData was a 2D array the values were loaded correctly in the file.

**Comment by SheetJSDev** [comment]
What is `csvData`?  Is it an array?  Can you share what `ws1` looks like?

**Comment by willofsound** [comment]
I don't know if this helps here or if someone searching for this will find it useful, but I was running into an issue with json_to_sheet where a lot of columns were missing when importing the data. While diving into the documentation, I found using json_to_sheet, there's an extra parameter that when not set, it will just ignore null/undefined values.

> **defval** - Use specified value in place of null or undefined 

So I used `XLSX.utils.sheet_to_json(ws, {defval:""})` to just place nothing in all my undefined value spots, which fixed the issue for my use.
